connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/arrow/ArrowEncoderSuite.scala

```diff
@@ -353,8 +355,20 @@ class ArrowEncoderSuite extends ConnectFunSuite with BeforeAndAfterAll {
 
   test("nullable fields") {
     val encoder = ScalaReflection.encoderFor[NullableData]
-    val instant = java.time.Instant.now()
-    val now = java.time.LocalDateTime.now()
+    // SPARK-44457: Similar to SPARK-42770, calling `truncatedTo(ChronoUnit.MICROS)`
+    // for `java.time.Instant` and `java.time.LocalDateTime` when testing with Java 17 on Linux.
+    val needTruncate =
```

Comment:
We can extract helper functions in a follow up and make `SQLImplicitsTestSuite` reuse the helper functions as well